src/train_model.py

```diff
@@ -41,8 +41,7 @@ def on_save_checkpoint(self, trainer, pl_module, checkpoint):
         self.num_ckpts += 1
 
         # Print GPU memory usage
-        print(run("nvidia-smi -q -d MEMORY", shell=True, capture_output=True).stdout)
-
+        print(torch.cuda.memory_summary())  # Prints per device
```

Comment:
Are you sure this will print to the .out file on supercomputing?

Reply:
Yep, I verified it on a multi-gpu tiny transformer run.
[ckpt_test_out.txt](https://github.com/DRAGNLabs/301r_retnet/files/14761173/ckpt_test_out.txt)